core/src/no/ntnu/tdt4240/asteroids/entity/World.java

```diff
@@ -102,16 +103,40 @@ private void spawnObstacles(int currentObstacles) {
     private Entity createObstacle() {
         Entity obstacle = EntityFactory.getInstance().createObstacle();
 
-        // TODO: Improve obstacle spawn position
         PositionComponent position = obstacle.getComponent(PositionComponent.class);
         DrawableComponent drawable = obstacle.getComponent(DrawableComponent.class);
 
-        int x = MathUtils.random(0, Gdx.graphics.getWidth());
-        int y = Gdx.graphics.getHeight() + drawable.region.getRegionHeight() / 2;
```

Comment:
Note how I offset the spawn position by getRegionHeight() / 2 so that the whole texture is outside the screen bounds.